server/segment_rpc.go

```diff
@@ -59,24 +59,30 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// oInfo will be non-nil if we need to send an updated net.OrchestratorInfo
-	// to the broadcaster
+	// oInfo will be non-nil if we need to send an updated net.OrchestratorInfo to the broadcaster
 	var oInfo *net.OrchestratorInfo
 
-	if err := orch.ProcessPayment(payment, segData.ManifestID); err != nil {
-		glog.Errorf("Error processing payment: %v", err)
+	if paymentError := orch.ProcessPayment(payment, segData.ManifestID); paymentError != nil {
 
+		acceptableErr, ok := paymentError.(core.AcceptableError)
+		if !ok || !acceptableErr.Acceptable() {
+			glog.Errorf("Unacceptable error occured processing payment: %v", paymentError.Error())
+			http.Error(w, fmt.Sprintf("Internal Server Error: %s", paymentError.Error()), http.StatusInternalServerError)
```

Comment:
`http.StatusBadRequest` is properly a better fit here (if we make the switch we can also remove the "Internal Server Error" prefix)

Reply:
Fixed in aceab05